src/core/contracts/registry.test.ts

```diff
@@ -407,16 +409,25 @@ describe("registry", () => {
       contractAddr = identityContract.address;
       const tx = await register(contractAddr, "Animaniacs");
       dsnpUserURI = await getURIFromRegisterTransaction(tx);
-      const signedMessage = await sign(msg);
-      sig = signedMessage.signature;
+      msg = createBroadcast(
+        convertDSNPUserURIToDSNPUserId(dsnpUserURI),
+        "https://fakeurl.org",
+        "0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef"
+      );
+      signedAnnouncement = await sign(msg);
+      sig = signedAnnouncement.signature;
     });
 
     afterAll(async () => {
       await revertHardhat(provider);
     });
 
     it("returns true if the signer is authorized for the given permissions", async () => {
-      await expect(isSignatureAuthorizedTo(sig, msg, dsnpUserURI, permAllowed)).toBeTruthy();
+      await expect(isSignatureAuthorizedTo(sig, msg, dsnpUserURI, permAllowed)).resolves.toBeTruthy();
+    });
+
+    it("returns true if provided a signed announcement", async () => {
+      await expect(isSignatureAuthorizedTo(sig, signedAnnouncement, dsnpUserURI, permAllowed)).resolves.toBeTruthy();
     });
 
```

Comment:
should there be a test for the InvalidAnnouncementParameterError?

Reply:
yep, good catch

Reply:
test added